icc-x-api/icc-patient-x-api.ts

```diff
@@ -693,18 +705,19 @@ export class IccPatientXApi extends iccPatientApi {
     })
   }
 
-  initEncryptionKeys(user: models.UserDto, pat: models.PatientDto) {
+  initEncryptionKeysForDelegates(
+    user: models.UserDto,
+    pat: models.PatientDto,
+    delegateIds: string[]
+  ): Promise<models.PatientDto> {
     const hcpId = user.healthcarePartyId || user.patientId
     return this.crypto.initEncryptionKeys(pat, hcpId!).then((eks) => {
       let promise = Promise.resolve(
         _.extend(pat, {
           encryptionKeys: eks.encryptionKeys,
         })
       )
-      ;(user.autoDelegations
-        ? (user.autoDelegations.all || []).concat(user.autoDelegations.medicalInformation || [])
-        : []
-      ).forEach(
+      delegateIds.forEach(
```

Comment:
I think you really can't do that. You should uion the delegateIds with the auto-delegations even if some  delegateIds have been set. The underlying assumption was always that the delegateIds where extra shares